metric_learn/mmc.py

```diff
@@ -89,10 +80,14 @@ def _process_inputs(self, X, constraints):
     
     # check to make sure that no two constrained vectors are identical
     a,b,c,d = constraints
-    ident = _vector_norm(X[a] - X[b]) > 1e-9
-    a, b = a[ident], b[ident]
-    ident = _vector_norm(X[c] - X[d]) > 1e-9
-    c, d = c[ident], d[ident]
+    no_ident = vector_norm(X[a] - X[b]) > 1e-9
+    a, b = a[no_ident], b[no_ident]
+    no_ident = vector_norm(X[c] - X[d]) > 1e-9
+    c, d = c[no_ident], d[no_ident]
+    if len(a) == 0:
+      raise RuntimeError('No similarity constraints given for MMC.')
+    if len(c) == 0:
+      raise RuntimeError('No dissimilarity constraints given for MMC.')
```

Comment:
I think a `ValueError` makes more sense here. We should also specify that no _non-trivial_ constraints were provided, otherwise the user might be confused.